src/rcc.rs

```diff
@@ -1,26 +1,341 @@
 //! Reset and Clock Control
 
-use stm32l4x6::{rcc, RCC};
+// TODO right now the various configure functions reach into rcc directly. This is bad. Add an
+// opaque CR member to RCC, and add methods to CR and BDCR. They should probably take clock source
+// variant arguments.
 
-use ::cmp;
+use stm32l4x6::{rcc, RCC, PWR};
 
 use common::Constrain;
 use time::Hertz;
 use flash::ACR;
 
 impl Constrain<Rcc> for RCC {
+    /// Create an RCC peripheral handle.
+    ///
+    /// Per 6.2, "The MSI is used as system clock source after startup from Reset, configured
+    /// at 4 MHz."
     fn constrain(self) -> Rcc {
+        // Enable write access to the BDCR
+        unsafe { (*PWR::ptr()).cr1.modify(|_,w| w.dbp().set_bit()); }
+        // Write access is (similarly) disabled in CFGR::freeze()
+        // TODO add PWR to the hal to avoid the above nastiness
         Rcc {
             ahb: AHB(()),
             apb1: APB1(()),
             apb2: APB2(()),
             bdcr: BDCR(()),
             csr: CSR(()),
-            cfgr: CFGR { hclk: None, pclk1: None, pclk2: None, sysclk: None }
+            cfgr: CFGR {
+                hclk: None,
+                pclk1: None,
+                pclk2: None,
+                sysclk: clocking::SysClkSource::MSI(
+                    clocking::MediumSpeedInternalRC::new(4_000_000, false)
+                    ),
+            }
         }
     }
 }
 
+pub mod clocking {
```

Comment:
I suggest to split rcc.rs into several files.
The structure can be something like
`src/rcc/mod.rs`
`src/rcc/clocking.rs`